packages/web3-core/src/web3_subscriptions.ts

```diff
@@ -17,45 +17,44 @@ along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
 
 // eslint-disable-next-line max-classes-per-file
 import {
-	HexString,
 	BlockOutput,
-	Web3BaseProvider,
-	Web3APISpec,
-	Web3APIParams,
+	DEFAULT_RETURN_FORMAT,
+	DataFormat,
 	EthExecutionAPI,
-	Log,
-	JsonRpcNotification,
 	JsonRpcSubscriptionResult,
-	DataFormat,
-	DEFAULT_RETURN_FORMAT,
 	JsonRpcSubscriptionResultOld,
-	EIP1193Provider,
+	JsonRpcNotification,
+	Log,
+	HexString,
+	Web3APIParams,
+	Web3APISpec,
 } from 'web3-types';
 import { jsonRpc } from 'web3-utils';
+// eslint-disable-next-line import/no-cycle
+import { Web3SubscriptionManager } from './web3_subscription_manager.js';
 import { Web3EventEmitter, Web3EventMap } from './web3_event_emitter.js';
 
-import { Web3RequestManager } from './web3_request_manager.js';
-
 export abstract class Web3Subscription<
 	EventMap extends Web3EventMap,
 	// eslint-disable-next-line @typescript-eslint/no-explicit-any
 	ArgsType = any,
 	API extends Web3APISpec = EthExecutionAPI,
 > extends Web3EventEmitter<EventMap> {
-	private readonly _requestManager: Web3RequestManager<API>;
+	private readonly _subscriptionManager: Web3SubscriptionManager<API>;
 	private readonly _lastBlock?: BlockOutput;
 	private readonly _returnFormat: DataFormat;
-	private _id?: HexString;
-	private _messageListener?: (data?: JsonRpcNotification<Log>) => void;
+	protected _id?: HexString;
 
 	public constructor(
 		public readonly args: ArgsType,
-
-		options: { requestManager: Web3RequestManager<API>; returnFormat?: DataFormat },
+		subscriptionManager: Web3SubscriptionManager,
```

Comment:
Thanks for sending me more info on this, As discussed this will break API so I suggest to post a fix without that.